presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -317,6 +355,17 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
                         Expression notAny = new NotExpression(rewrittenAny);
                         subPlan.getTranslations().addIntermediateMapping(quantifiedComparison, notAny);
                         return appendQuantifiedComparisonApplyNode(subPlan, rewrittenAny, correlationAllowed);
+                    case ANY:
+                    case SOME:
+                        // A <> ANY B <=> min B <> max B || A <> min B <=> !(min B = max B && A = min B) <=> !(A = ALL B)
```

Comment:
Well, that's what it essentially does, because ... <=>!(A = ALL B) <=> !(A IN B).
However, for an implementation I'm reusing (A=ALL B) case, so my equation stops at that point.